src/Pages/ContentListTable/ContentListTable.tsx

```diff
@@ -240,68 +241,101 @@ const ContentListTable = () => {
   } = data;
 
   const rowActions = useCallback(
-    (rowData: ContentItem): IAction[] =>
-      isRedHatRepository
-        ? features?.snapshots?.accessible
-          ? [
+    (rowData: ContentItem): IAction[] => {
+      const triggerSnapshotAction = {
+        isDisabled: actionTakingPlace || rowData?.status === 'Retrying',
+        title: 'Trigger Snapshot',
+        onClick: async () => {
+          try {
+            await axios.post(`/api/snapshot/${rowData?.uuid}/trigger`);
+          } catch (error) {
+            console.error('Error triggering snapshot:', error);
+          }
+        },
+      };
+
+      const express = require('express');
+      const axios = require('axios');
+
+      const app = express();
+      const PORT = process.env.PORT || 3000;
+      const introspectRepository = async (request) => {
+        const { data } = await axios.post(
+          `/api/content-sources/v1/repositories/${request.uuid}/introspect/`,
+          { reset_count: request.reset_count },
+        );
+        return data;
+      };
+      app.post('/trigger-snapshot', async (req, res) => {
+        try {
+          const request = {
+            uuid: 'your_repository_uuid_here',
+            reset_count: true,
+          };
+          const snapshotData = await introspectRepository(request);
+          res.status(200).json({ message: 'Snapshot triggered successfully', data: snapshotData });
+        } catch (error) {
+          const errorMessage = error instanceof Error ? error.message : 'Unknown error occurred';
+          res.status(500).json({ message: 'Error triggering snapshot', error: errorMessage });
+        }
+      });
+      app.listen(PORT, () => {
+        console.log(`Server is running on port ${PORT}`);
+      });
```

Comment:
The frontend does not need to run a server to listen for an api call, that is what the backend is for, its already listening.